internal/k8s/ingress_link.go

```diff
@@ -0,0 +1,124 @@
+package k8s
+
+import (
+	"github.com/golang/glog"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/tools/cache"
+)
+
+func createIngressLinkHandlers(lbc *LoadBalancerController) cache.ResourceEventHandlerFuncs {
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			link := obj.(*unstructured.Unstructured)
+			glog.V(3).Infof("Adding IngressLink: %v", link.GetName())
+			lbc.AddSyncQueue(link)
+		},
+		DeleteFunc: func(obj interface{}) {
+			link, isUnstructured := obj.(*unstructured.Unstructured)
+
+			if !isUnstructured {
+				deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+				if !ok {
+					glog.V(3).Infof("Error received unexpected object: %v", obj)
+					return
+				}
+				link, ok = deletedState.Obj.(*unstructured.Unstructured)
+				if !ok {
+					glog.V(3).Infof("Error DeletedFinalStateUnknown contained non-Unstructured object: %v", deletedState.Obj)
+					return
+				}
+			}
+
+			glog.V(3).Infof("Removing IngressLink: %v", link.GetName())
+			lbc.AddSyncQueue(link)
+		},
+		UpdateFunc: func(old, cur interface{}) {
+			oldLink := old.(*unstructured.Unstructured)
+			curLink := cur.(*unstructured.Unstructured)
+			different, err := areResourcesDifferent(oldLink, curLink)
+			if err != nil {
+				glog.V(3).Infof("Error when comparing IngressLinks: %v", err)
+				lbc.AddSyncQueue(curLink)
+			}
+			if different {
+				glog.V(3).Infof("IngressLink %v changed, syncing", oldLink.GetName())
+				lbc.AddSyncQueue(curLink)
+			}
+		},
+	}
+}
+
+func (lbc *LoadBalancerController) addIngressLinkHandler(handlers cache.ResourceEventHandlerFuncs, name string) {
+	optionsModifier := func(options *meta_v1.ListOptions) {
+		options.FieldSelector = fields.Set{"metadata.name": name}.String()
+	}
+
+	informer := dynamicinformer.NewFilteredDynamicInformer(lbc.dynClient, ingressLinkGVR, lbc.controllerNamespace, lbc.resync,
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}, optionsModifier)
+
+	informer.Informer().AddEventHandlerWithResyncPeriod(handlers, lbc.resync) //nolint:errcheck,gosec
```

Comment:
is `//nolint:errcheck,gosec` needed here?

Reply:
to suppress lint issues with not handling the potential error returned from `AddEventHandlerWithResyncPeriod`